### PR TITLE
Feature/add is loading as parameter

### PR DESCRIPTION
**Pull request by hiagolcm**
Add `isLoading` as a `ProductSummaryProvider` parameter. 

[workspace](https://hiago--carrefourbrfood.myvtex.com/Alimentos-e-Bebidas/Mercearia-e-Alimentos-Basicos?crfint=hm-tlink|alimentos-e-bebidas|1|mercearia|4)

### CODE REVIEW

File: react/ProductSummaryContext.tsx [hunk 1]
```diff
@@ -115,11 +115,11 @@ const buildProductQuery = ((product: Product) => {
   return querystring.stringify(query)
 })
 
-function ProductSummaryProvider({ product, selectedItem, children }) {
+function ProductSummaryProvider({ product, selectedItem, isLoading, children }) {
   const initialState = {
     product,
     isHovering: false,
-    isLoading: false,
+    isLoading: Boolean(isLoading),
```
Comment (iaronaraujo) [inline react/ProductSummaryContext.tsx]: Why is this necessary? Can't you just pass the `isLoading` value?

Comment (thalytafabrine) [inline react/ProductSummaryContext.tsx]: just a suggestion, once I don't know if you really need the `Boolean()` 🤔 

```suggestion
    isLoading: isLoading ?? false,
```

File: react/ProductSummaryContext.tsx [hunk 2]
```diff
@@ -115,11 +115,11 @@ const buildProductQuery = ((product: Product) => {
   return querystring.stringify(query)
 })
 
-function ProductSummaryProvider({ product, selectedItem, children }) {
+function ProductSummaryProvider({ product, selectedItem, isLoading, children }) {
```
Comment (klzns) [inline react/ProductSummaryContext.tsx]: Adding a default value just to be backwards compatible

```suggestion
function ProductSummaryProvider({ product, selectedItem, isLoading = false, children }) {
```

**Review by klzns — CHANGES_REQUESTED** [review]

File: react/ProductSummaryContext.tsx (hunk 1 above)
Comment (hiagolcm) [reply react/ProductSummaryContext.tsx]: It is not necessary 🤦 
Removed!

File: react/ProductSummaryContext.tsx (hunk 2 above)
Comment (hiagolcm) [reply react/ProductSummaryContext.tsx]: Done!

File: react/ProductSummaryContext.tsx (hunk 1 above)
Comment (hiagolcm) [reply react/ProductSummaryContext.tsx]: It was not necessary 🤦

**Review by iaronaraujo — APPROVED** [review]

File: react/ProductSummaryContext.tsx [hunk 3]
```diff
@@ -115,14 +135,15 @@ const buildProductQuery = ((product: Product) => {
   return querystring.stringify(query)
 })
 
-function ProductSummaryProvider({ product, selectedItem, children }) {
+function ProductSummaryProvider({ product, selectedItem, isLoading=false, children }) {
   const initialState = {
     product,
     isHovering: false,
-    isLoading: false,
+    isLoading: isLoading,
```
Comment (thalytafabrine) [inline react/ProductSummaryContext.tsx]:
```suggestion
    isLoading,
```

File: react/ProductSummaryContext.tsx [hunk 4]
```diff
@@ -150,27 +171,15 @@ function ProductSummaryProvider({ product, selectedItem, children }) {
 }
 
 function useProductSummaryDispatch() {
-  const context = useContext(ProductDispatchContext)
-
-  if (context === undefined) {
-    throw new Error('useProductSummaryDispatch must be used within a ProductSummaryDispatchProvider')
-  }
-
```
Comment (thalytafabrine) [inline react/ProductSummaryContext.tsx:158]: why did you remove this?

Comment (hiagolcm) [reply react/ProductSummaryContext.tsx:158]: Some components may be in both PDP or product-summary. If one of those components uses this context, it will work on product-summary but break on PDP.
